### DISCUSSION TITLE
HmIP-ESI erweiterte Verbrauchszähler in HA verfügbar machen

### DISCUSSION THREAD

**Discussion by hksthff** [OpenCCU/OpenCCU#2839] [🇩🇪 Fragen&Antworten] [answered]
Moin, 

habe seit ein paar Tagen den HmIP-ESI im Betrieb und seit heute per PIN auch die erweiterten Werte freischalten können - werden auch auf Anhieb in RaspberryMatic angezeigt! 

<img width="546" alt="screen1" src="https://github.com/user-attachments/assets/f4a2ec76-8c2b-4a52-9639-a1fd81cb7b42">

Leider kommen bei HA aber nur die drei Standardwerte vom HmIP-ESI an:

<img width="1137" alt="screen2" src="https://github.com/user-attachments/assets/af8ee378-8c73-4746-bb33-ccb8fdbc97f3">

Ich habe in diesem Beitrag gelesen (https://homematic-forum.de/forum/viewtopic.php?f=31&t=82352), dass die erweiterten Werte vom HmIP-ESI nicht in Systemvariablen geschrieben werden, sondern nur als Metadaten.

Da die Werte in RaspberryMatic aber bereits vorliegen (siehe Screenshot) - gibt es eine Möglichkeit, die auch als Entität in HA verfügbar zu machen?

Danke vorab und beste Grüße!

**Comment by Baxxy13** [comment, marked as answer]
Keine der beiden möglichen Integrationen erzeugt aus Metadaten Entitäten. 
Entweder du gehst den Umweg über die Systemvariablen oder du legst dir in HA einfach mehrere "Verbrauchszähler-Helfer" an. 
Ich würde letzteres machen.

**Reply by hksthff** [reply]
Danke dir, bin jetzt bei den Verbrauchszählerhelfern eingestiegen und probiere darüber eine brauchbare Statistik aufzubauen. Falls andere da später auch noch drüber stolpern sollten: habe hier einen guten Einstieg gefunden: https://www.bjoerns-techblog.de/2024/06/hmip-esi-iec-in-home-assistant/.
